db/virtual_table.hh

```diff
@@ -55,4 +56,16 @@ public:
     void set_database(database& db) { _db = &db; }
 };
 
+// Produces results by filling a memtable on each read.
+// Use when the amount of data is not significant relative to shard's memory size.
+class memtable_filling_virtual_table : public virtual_table {
+public:
+    using virtual_table::virtual_table;
+
+    virtual future<> execute(memtable& mt, std::function<void(size_t)> update_permit) = 0;
```

Comment:
This API has some drawbacks which can be avoided.

First thing is that with this the user needs to have a discipline of calling update_permit(). It's not enforced.

Second, is that the user needs to know how to do it properly.

The two can be solved by hiding permit and memtable management inside `memtable_filling_virtual_table` implementation and only expose a mutation sink to the execute() implementation:
```
virtual future<> execute(std::function<void(mutation)> mutation_sink) = 0;
```